File: Algorithm/String/6-make-it-anagram.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

int main(int argc, char const *argv[])
{
    string one;
    string two;
    int count = 0;

    cin >> one >> two;

    int some[26] = {0};
    int other[26] = {0};

    for (int i = 0; i < one.length(); ++i)
    {
        some[one[i] - 'a'] ++;
    }

    for (int i = 0; i < two.length(); ++i)
    {
        other[two[i] - 'a'] ++;
    }

    for (int i = 0; i < 26; ++i)
    {
        count += abs(some[i] - other[i]);
    }

    cout << count << endl;

    return 0;
}

File: Algorithm/String/6-make-it-anagram.php
<?php

$handle = fopen("php://stdin", "r");
fscanf($handle, "%s", $one);
fscanf($handle, "%s", $two);

$some = [];
$other = [];

for ($i=0; $i < 26; $i++) {
    $some[$i] = 0;
    $other[$i] = 0;
}

for ($i=0; $i < strlen($one); $i++) {
    $index = ord($one[$i]) - ord('a');

    if (isset($some[$index])) {
        $some[$index] ++;
    } else{
        $some[$index] = 0;
    }
}

for ($i=0; $i < strlen($two); $i++) {
    $index = ord($two[$i]) - ord('a');

    $other[$index] ++;
}

$count = 0;
for ($i=0; $i < 26; $i++) {
    $count += abs($some[$i] - $other[$i]);
}

echo $count . PHP_EOL;
